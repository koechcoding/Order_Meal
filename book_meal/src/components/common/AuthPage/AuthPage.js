import React from 'react';
import PropTypes from 'prop-types';
import Loading from 'react-loading-bar';
import 'react-loading-bar/dist/index.css';
import './styles.css';

const AuthPage = ({ styles = {}, children, loading}) =>(
    <main style={{...styles}} className="patterns bg-teal">
        <Loading show={loading} color="orange"/>
        <section className='container login'>
            <div className='row pt-5'>
                <div className='col-12 text-center'>
                    <h1 className='logo-small text-flav'>BAM!</h1>
                </div>
            </div>
            <div className='row mb-5 pl-3 pr-3 p-sm-0'>
                {children}
            </div>
        </section>
    </main>
);

AuthPage.propTypes = {
    children: PropTypes.element,
    loading: PropTypes.bool,
}

export default AuthPage;