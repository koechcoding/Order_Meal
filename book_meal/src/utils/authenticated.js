export default function authenticated(){
    try{
        return JSON.parse(localStorage.getItem('user'));
    } catch(err){
        
    }
}