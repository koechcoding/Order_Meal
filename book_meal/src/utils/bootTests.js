import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
Enzyme.configure({ adapter: new Adapter()});

global.window = {}
import localStorage from 'mock-local-Storage';
window.localStorage = global.localStorage

export default Enzyme;
