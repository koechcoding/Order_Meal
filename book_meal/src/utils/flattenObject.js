export default function flattenObject(obj){
    let result = {}
    for (const key in obj){
        if((typeof obj[key])=== 'object' &&  obj[key]  !== null) {
            let flattened = flattenObject(obj[key])
            for(const key2 in flattened){
                result[key + "." + key2] = flattened[key2];
            }
        }else{
            result[key] = obj[key];
        }
    }
    return result;
}