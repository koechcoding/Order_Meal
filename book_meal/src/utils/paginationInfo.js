export default function paginationInfo(page){

    if(!page){
        return{}
    }

    let currentPage = 1
    if(page.next_page)
         currentPage = page.next_page -1;
    else if (page.prev_page)
        currentPage = page.prev_page + 1;

    let pageCount = Math.ceil(page.total / page.per_paage);
    if(Number.isNaN(pageCount) || pageCount === 0)
      pageCount = 1; 

    return {
        pageCount,
        currentPage,
        total: page.total,
        perPage: page.per_page,
        hasNext: page.has_next,
        hasPrev: page.has_prev,
        nextPage: page.next_page,
        prevPage: page.prev_page,
        currentCount: page.current_count,
    };
}