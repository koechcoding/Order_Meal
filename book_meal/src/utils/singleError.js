const singleError = (errorPayload)=>{
    if(errorPayload && errorPayload.data){
        const response = errorPayload.data;
        //alternative to Object.keys...
        for(let errorKey in response.errors){
            return response.error[errorKey]
        }
        if(response.message){
            return response.message;
        }
    }
    return 'Unknown error. Please try again!';
}
export default singleError;