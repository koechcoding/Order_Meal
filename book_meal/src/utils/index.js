import singleError from './singleError';
import paginationInfo from './paginationInfo';
import authenticated from './authenticated';
import flattenObject from './flattenObject';

export { 
    singleError, 
    paginationInfo, 
    authenticated,
    flattenObject
}