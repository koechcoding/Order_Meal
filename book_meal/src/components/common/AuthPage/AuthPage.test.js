import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import 'src/utils/bootTests'; //order matters
import AuthPage from './AuthPage';
import Adapter from 'enzyme-adapter-react-16';

describe('<AuthPage />', ()=>{
    it('renders correctly', ()=>{
        const wrapper = shallow(<AuthPage/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.logo-small').text()).toEqual('BAM!')
    });
});