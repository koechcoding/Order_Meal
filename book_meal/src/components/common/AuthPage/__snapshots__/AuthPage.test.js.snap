// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthPage /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AuthPage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Loading
          change={true}
          color="orange"
          show={undefined}
          showSpinner={true}
/>,
        <section
          className="container login"
>
          <div
                    className="row pt-5"
          >
                    <div
                              className="col-12 text-center"
                    >
                              <h1
                                        className="logo-small text-flav"
                              >
                                        BAM!
                              </h1>
                    </div>
          </div>
          <div
                    className="row mb-5 pl-3 pr-3 p-sm-0"
          />
</section>,
      ],
      "className": "patterns bg-teal",
      "style": Object {},
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "change": true,
          "color": "orange",
          "show": undefined,
          "showSpinner": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="row pt-5"
>
              <div
                            className="col-12 text-center"
              >
                            <h1
                                          className="logo-small text-flav"
                            >
                                          BAM!
                            </h1>
              </div>
</div>,
            <div
              className="row mb-5 pl-3 pr-3 p-sm-0"
/>,
          ],
          "className": "container login",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="col-12 text-center"
>
                <h1
                                className="logo-small text-flav"
                >
                                BAM!
                </h1>
</div>,
              "className": "row pt-5",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <h1
                  className="logo-small text-flav"
>
                  BAM!
</h1>,
                "className": "col-12 text-center",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "BAM!",
                  "className": "logo-small text-flav",
                },
                "ref": null,
                "rendered": "BAM!",
                "type": "h1",
              },
              "type": "div",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "row mb-5 pl-3 pr-3 p-sm-0",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "section",
      },
    ],
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Loading
            change={true}
            color="orange"
            show={undefined}
            showSpinner={true}
/>,
          <section
            className="container login"
>
            <div
                        className="row pt-5"
            >
                        <div
                                    className="col-12 text-center"
                        >
                                    <h1
                                                className="logo-small text-flav"
                                    >
                                                BAM!
                                    </h1>
                        </div>
            </div>
            <div
                        className="row mb-5 pl-3 pr-3 p-sm-0"
            />
</section>,
        ],
        "className": "patterns bg-teal",
        "style": Object {},
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "change": true,
            "color": "orange",
            "show": undefined,
            "showSpinner": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="row pt-5"
>
                <div
                                className="col-12 text-center"
                >
                                <h1
                                                className="logo-small text-flav"
                                >
                                                BAM!
                                </h1>
                </div>
</div>,
              <div
                className="row mb-5 pl-3 pr-3 p-sm-0"
/>,
            ],
            "className": "container login",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="col-12 text-center"
>
                  <h1
                                    className="logo-small text-flav"
                  >
                                    BAM!
                  </h1>
</div>,
                "className": "row pt-5",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <h1
                    className="logo-small text-flav"
>
                    BAM!
</h1>,
                  "className": "col-12 text-center",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "BAM!",
                    "className": "logo-small text-flav",
                  },
                  "ref": null,
                  "rendered": "BAM!",
                  "type": "h1",
                },
                "type": "div",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "row mb-5 pl-3 pr-3 p-sm-0",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "section",
        },
      ],
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;